### PR TITLE
Add Escape Uri String

### PR DESCRIPTION
**Pull request by MattiLE**
Hi al-team, 

I have added one more function to the Uri codeunit.
Actually only the C# function EscapeUriString is called. https://docs.microsoft.com/en-us/dotnet/api/system.uri.escapeuristring?view=net-5.0

Can you include the customization in the base version?

### CODE REVIEW

**Comment by bjarkihall** [comment]
Usually you'd want to use EscapeDataString, which Uri codeunit already provides:
https://github.com/microsoft/ALAppExtensions/blob/2109d868be60047a3db66b223e453239add5c269/Modules/System/URI/src/Uri.Codeunit.al#L115

Here's the discussion about it:
https://stackoverflow.com/questions/4396598/whats-the-difference-between-escapeuristring-and-escapedatastring

I'm a bit curious about the motivation for the PR as I think the difference between the two procedures is pretty confusing.

File: Modules/System/URI/src/Uri.Codeunit.al [hunk 1]
```diff
@@ -128,6 +128,17 @@ codeunit 3060 Uri
         exit(Uri.UnescapeDataString(TextToUnescape));
     end;
 
+    /// <summary>
+    /// Converts a uri string to its escaped representation.
+    /// </summary>
+    /// <remarks>Visit https://docs.microsoft.com/en-us/dotnet/api/system.uri.escapeuristring for more information.</remarks>
+    /// <param name="TextToEscape"></param>
```
Comment (mazhelez) [inline Modules/System/URI/src/Uri.Codeunit.al:135]: Doc is missing.

File: Modules/System/URI/src/Uri.Codeunit.al [hunk 2]
```diff
@@ -128,6 +128,17 @@ codeunit 3060 Uri
...
+    /// <returns>A string that contains the escaped representation of <paramref name="TextToEscape:"/>.</returns>
```
Comment (mazhelez) [inline Modules/System/URI/src/Uri.Codeunit.al:136]:
```suggestion
    /// <returns>A string that contains the escaped representation of <paramref name="TextToEscape"/>.</returns>
```

**Review by mazhelez — CHANGES_REQUESTED** [review]
Looks okay.

A couple of minor things need fixing. Maybe a test or two would be great.

**Comment by MattiLE** [comment]
Hi @bjarkihall, background of this PR is that i have created a URL Combine function. A URL is to be built from separate parts.
At the end of the function, all characters that do not belong to a URL should be escaped. Therefore, if the URL does not contain "%", I can simply use the EscpaeURIString method. https://stackoverflow.com/a/47636037/62600

However, you are right that it can be confusing if both functions are included.

**Comment by mazhelez** [comment]
Hey @MattiLE 

Do you plan on making another iteration? We'd love to uptake your PR, however, there're still comments that need to be fixed.

**Comment by MattiLE** [comment]
Hi @mazhelez, 

it sounded to me that PR was being rejected. I might have misunderstood @bjarkihall.
I will now implement the annotations :)

**Comment by bjarkihall** [comment]
Sorry @MattiLE I don't have any authority here, I was just curious for the motivation of the PR as there are better alternatives already present in BC.

For your Url combine function, you can have a look at the latest changes: https://github.com/microsoft/ALAppExtensions/tree/master/Modules/System/URI/src - now all of the uri properties are exposed: scheme, host, port, path, query, fragment.

Maybe it would make sense to move the HttpUtility wrappers from Base app (Type Helper codeunit) to System app (Uri module) as well or overload the constructor, similar to what you suggested in #7551. Just a thought.

I just want to make sure the API doesn't get out of hand, since these functions are already confusing most .NET programmers and I haven't heard very good things about EscapeUriString, but the decision itself is up to MS of course.

**Comment by mazhelez** [comment]
@MattiLE It's an open discussion :)

The idea is to contribute something that you and other AL developers might potentially us

**Comment by mazhelez** [comment]
@MattiLE Do you plan on working on this PR? If not, can you please close it?

**Comment by MattiLE** [comment]
Hi @mazhelez, I can't say when I will find time to work on this further. I'll close the PR for now and might do a new one later. :)

**Pull request closed without merging**